informer/informer.go

```diff
@@ -100,11 +100,13 @@ func New(config Config) (*Informer, error) {
 		resyncPeriod: config.ResyncPeriod,
 	}
 
-	prometheus.MustRegister(prometheus.Collector(newInformer))
-
 	return newInformer, nil
 }
 
+func (i *Informer) Boot() {
```

Comment:
I'd like our boot methods to take context and return error. Like `Bootl(ctx context.Context) error`.

- `ctx` to be able to cancel.
- `error` for unified interface. Can be useful at times. We always can ignore it.

Reply:
I agree except ignoring errors. We must never ignore them. This will fuck us pretty bad in the worst possible ways and then Tim will lose all his shit. I see it already. :D

Reply:
I wouldn't like to ignore errors at all, but I saw `go controller.Boot()` so many times I lost my shit already. But we can always wrap them with anonymous panicking function (as quick solution) and think what to do with that.

Reply:
PTAL if this is fine now